sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,299 @@
+= sycl_ext_oneapi_group_load_store
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+
+IMPORTANT: This specification is a draft.
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 6.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort] (`group_with_scratchpad` class used as GroupHelper).
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Overview
+
+This extension defines free functions for load/store operations within work
+group scope.
+
+== Single Value Functions
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API scalar
+template<typename Group, typename InputIteratorT,
+         typename OutputT, typename Properties = properties<>>
+void group_load(Group g, InputIteratorT in_ptr,
+                      OutputT& out, Properties = {}); // (1)
+
+// Store API scalar
+template<typename Group, typename InputT,
+         typename OutputIteratorT, typename Properties = properties<>>
+void group_store(Group g, const InputT& in,
+                      OutputIteratorT out_ptr, Properties = {}); // (2)
+
+}
+----
+
+(1) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `Group` is a work-group or sub-group.
```

Comment:
I think we may want to add some additional constraints on `InputIteratorT` and `OutputT` here. 

For example, do we require `InputIteratorT` to be [random access iterator](https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator) or simple [forward iterator](https://en.cppreference.com/w/cpp/named_req/ForwardIterator) should be enough?

Do we require type `T` which is obtained by dereferencing iterator of type `InputIteratorT` to be exactly the same as `OutputT`? Or is it enough if `T` is convertible to `OutputT`?

The same comment applies to all other functions and overloads proposed in this extension.

Reply:
Updated, please, check if the wording is ok